core/quantity/src/Parser.ts

```diff
@@ -333,22 +333,33 @@ export class Parser {
           }
         } else {
           // not processing a number
-          if (isOperator(charCode)) {
+          const isCharOperator = isOperator(charCode);
+          const isSpacer = charCode === format.spacerOrDefault.charCodeAt(0) && charCode !== QuantityConstants.CHAR_SPACE;
+
+          if (isSpacer && i > 0 && i < str.length - 1) {
+            const prevCharCode = str.charCodeAt(i - 1);
+            const nextCharCode = str.charCodeAt(i + 1);
+            if (isCharOperator && nextCharCode !== QuantityConstants.CHAR_SPACE && prevCharCode !== QuantityConstants.CHAR_SPACE) {
+              // ignore spacer if it's not at the start or end, not whitespace, and is not in between whitespace
+              continue;
```

Comment:
I don't think we want to force a white space on either side of the operator.

If we are going to rely on spacing maybe a space before the operator is enough.

Another idea that is more complex is that we know how many segments there could be so we know the max number of separators ... if we see more it must be an operator.  If we did this we would still need to do something with spaces to handle cases where a value didn't need all segments.  e.g. exactly 12'.

I'd be happy to start simple and say there must be a space before if the spacer character matches the operator.

Reply:
alrighty, i've edited the rule so just having a space before the operator is enough. no regression in our tests, I also added a few test cases where following the operator is a non white space character too.